bundles/org.openhab.core.model.persistence/src/org/openhab/core/model/persistence/internal/PersistenceModelManager.java

```diff
@@ -47,75 +59,55 @@
  *
  * @author Kai Kreuzer - Initial contribution
  * @author Markus Rathgeb - Move non-model logic to core.persistence
+ * @author Jan N. Klug - Refactored to {@link PersistenceServiceConfigurationProvider}
  */
-@Component(immediate = true)
-public class PersistenceModelManager implements ModelRepositoryChangeListener {
-
-    private ModelRepository modelRepository;
-
-    private PersistenceManager manager;
-
-    public PersistenceModelManager() {
-    }
+@Component(immediate = true, service = PersistenceServiceConfigurationProvider.class)
+@NonNullByDefault
+public class PersistenceModelManager extends AbstractProvider<PersistenceServiceConfiguration>
+        implements ModelRepositoryChangeListener, PersistenceServiceConfigurationProvider {
+    private final Map<String, PersistenceServiceConfiguration> configurations = new ConcurrentHashMap<>();
+    private final ModelRepository modelRepository;
+
+    @Activate
+    public PersistenceModelManager(@Reference ModelRepository modelRepository) {
+        this.modelRepository = modelRepository;
 
-    protected void activate() {
         modelRepository.addModelRepositoryChangeListener(this);
-        for (String modelName : modelRepository.getAllModelNamesOfType("persist")) {
-            addModel(modelName);
-        }
+        modelRepository.getAllModelNamesOfType("persist")
+                .forEach(modelName -> modelChanged(modelName, EventType.ADDED));
     }
 
+    @Deactivate
     protected void deactivate() {
         modelRepository.removeModelRepositoryChangeListener(this);
-        for (String modelName : modelRepository.getAllModelNamesOfType("persist")) {
-            removeModel(modelName);
```

Comment:
Technically it is not needed as long as the registry is the only listener, because on de-activation the provider is removed from the registry and the implementation in `AbstractRegistry` then removes all elements that were provided by the removed provider.

But if you prefer to remove it anyway, I can re-add it.

Reply:
After thinking about it, I believe that it is cleaner if we remove it ourselves.